modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRow.es.js

```diff
@@ -0,0 +1,326 @@
+			[
+				<span key="model-name">
+					{modelLabel}
+				</span>,
+				<b key="property">
+					{propertyLabel}
+				</b>,
+				<span key="operator">
+					{operatorLabel}
+				</span>,
+				<b key="value">
+					{value}
+				</b>
+			],
+			false
+		);
+	}
+
+	_getSelectedItem = (list, idSelected) =>
+		list.find(item => item.name === idSelected);
+
+	_handleDelete = event => {
+		event.preventDefault();
+
+		const {index, onDelete} = this.props;
+
+		onDelete(index);
+	}
+
+	_handleDuplicate = event => {
+		event.preventDefault();
+
+		const {criterion, index, onAdd} = this.props;
+
+		onAdd(index + 1, criterion);
+	}
+
+	_handleInputChange = propertyName => event => {
+		const {criterion, onChange} = this.props;
+
+		onChange(
+			{
+				...criterion,
+				[propertyName]: event.target.value
+			}
+		);
+	};
+
+	render() {
+		const {
+			canDrop,
+			connectDragPreview,
+			connectDragSource,
+			connectDropTarget,
+			criterion,
+			dragging,
+			editing,
+			hover,
+			modelLabel,
+			supportedOperators,
+			supportedProperties,
+			supportedPropertyTypes
+		} = this.props;
+
+		const selectedOperator = this._getSelectedItem(
+			supportedOperators,
+			criterion.operatorName
+		);
+
+		const selectedProperty = this._getSelectedItem(
+			supportedProperties,
+			criterion.propertyName
+		);
+
+		const operatorLabel = selectedOperator ? selectedOperator.label : '';
+		const propertyLabel = selectedProperty ? selectedProperty.label : '';
+		const value = criterion ? criterion.value : '';
+
+		const propertyType = selectedProperty ? selectedProperty.type : '';
+
+		const filteredSupportedOperators = supportedOperators.filter(
+			operator =>
+				supportedPropertyTypes[propertyType].includes(operator.name)
+		);
+
+		const classes = getCN(
+			'criterion-row-root',
+			{
+				'dnd-drag': dragging,
+				'dnd-hover': hover && canDrop
+			}
+		);
+
+		return connectDropTarget(
+			connectDragPreview(
+				<div
+					className={classes}
+				>
+					{editing ? (
+						<div className="edit-container">
+							{connectDragSource(
+								<div className="drag-icon">
+									<ClayIcon iconName="drag" />
+								</div>
+							)}
+
+							<span className="criterion-string">
+								{sub(
+									Liferay.Language.get('x-with-property-x'),
+									[
+										<span key="model-name">
+											{modelLabel}
+										</span>,
+										<b key="property">
+											{propertyLabel}
+										</b>
+									],
+									false
+								)}
+							</span>
+
+							<ClaySelect
+								className="criterion-input operator-input form-control"
+								onChange={this._handleInputChange(
+									'operatorName'
+								)}
+								options={filteredSupportedOperators.map(
+									({label, name}) => ({
+										label,
+										value: name
+									})
+								)}
+								selected={selectedOperator && selectedOperator.name}
+							/>
+
+							<input
+								className="criterion-input form-control"
+								id="queryRowValue"
```

Comment:
What's this for? If it's necessary, make sure you namespace it with the portlet namespace.

Reply:
This isn't getting submitted, so the id isn't necessary. I'll remove it.